src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1487,8 +1483,10 @@ def enable_features(cmd, client, resource_group_name, cluster_name, features, ku
         cmd_helm_upgrade.extend(["--kube-context", kube_context])
     if enable_azure_rbac:
         cmd_helm_upgrade.extend(["--set", "systemDefaultValues.guard.enabled=true"])
-        cmd_helm_upgrade.extend(["--set", "systemDefaultValues.guard.clientId={}".format(azrbac_client_id)])
-        cmd_helm_upgrade.extend(["--set", "systemDefaultValues.guard.clientSecret={}".format(azrbac_client_secret)])
+        # Setting the default authnMode mode as "arc" for guard. This mode used Arc RBAC 1P apps for authN/authZ.
+        # This mode in guard uses PoP token based auth.
+        cmd_helm_upgrade.extend(["--set", "systemDefaultValues.guard.authnMode=arc"])
```

Comment:
Can this be defaulted on the chart side ?

Reply:
If we default this on the chart size, will there be any implications for customers who are using RBAC in 3P mode and have agent auto upgrade to true? Can we do it in phases?
@julienstroheker

Reply:
If you change the default behavior, you need to confirm the impact of the breaking change on the customers.
We usually recommend implementing the breaking change in the breaking change window (Ignite and Build sprint), and before that, we suggest recommending customers to change their behavior through notification messages (warning log)

Reply:
Leaving the defaults as-is for now and will revisit later